engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSource.scala

```diff
@@ -18,7 +20,11 @@ import scala.reflect._
 trait FlinkSource[T] extends Source[T] {
 
   def sourceStream(env: StreamExecutionEnvironment,
-                   flinkNodeContext: FlinkCustomNodeContext): DataStream[T]
+                   flinkNodeContext: FlinkCustomNodeContext): DataStream[Context]
+
+}
+
+trait SourceTestSupport[T] { self: Source[T] =>
```

Comment:
Maybe we can just make this trait extend ```TestDataParserProvider```? In usage you require it anyway. And it would be more clear you need both to make stuff work?